cassandra/cluster.py

```diff
@@ -3612,6 +3608,7 @@ def _set_result(self, host, connection, pool, response):
                 # we got some other kind of response message
                 msg = "Got unexpected message: %r" % (response,)
                 exc = ConnectionException(msg, host)
+                self._cancel_timer()
```

Comment:
The goal is to cancel as quickly as possible -- without waiting for callbacks to execute in `defunct` below -- to avoid a pileup of timers when clients don't back off requests on a failing connection as described in [PYTHON-367](https://datastax-oss.atlassian.net/browse/PYTHON-367). Do you think this accomplishes that goal?